script.js

```diff
@@ -14,30 +12,122 @@ function createCustomElement(element, className, innerText) {
-  return item.querySelector('span.item__sku').innerText;
+function cartItemClickListener(event) { // Removing cart item by clicking it
+  event.target.remove();
+  toLocalStorage();
+  totalPrice();
 }
 
-function cartItemClickListener(event) {
-  // coloque seu código aqui
-}
+const toGetLocalStorage = () => { // Get saved list from local storage
+  const getList = localStorage.getItem('cart_products');
+  document.querySelector('.cart__items').innerHTML = getList; // Adding items to current list
+  const cart = document.querySelector('.cart__items'); // Making the items clickable again
+  cart.addEventListener('click', cartItemClickListener);
+};
 
-function createCartItemElement({ sku, name, salePrice }) {
+const emptyCart = () => { // Function to delete all items form the cart at once
+  const emptyCartBtt = document.querySelector('.empty-cart');
+  emptyCartBtt.addEventListener('click', () => {
+    const itemsList = document.querySelector('.cart__items');
+    itemsList.innerHTML = '';
+    toLocalStorage();
+    totalPrice();
+  });
+};
+
+function createCartItemElement({ sku, name, salePrice }) { // Function to insert item in the cart
   const li = document.createElement('li');
   li.className = 'cart__item';
   li.innerText = `SKU: ${sku} | NAME: ${name} | PRICE: $${salePrice}`;
   li.addEventListener('click', cartItemClickListener);
   return li;
 }
+
+// Function to create the item object as required, with sku, name and salePrice
+const newCartItem = (itemObj) => {
+  const cartItems = document.querySelector('.cart__items');
+  const { id, title, price } = itemObj;
+  const item = {
+    sku: id,
+    name: title,
+    salePrice: price,
+  };
+  cartItems.appendChild(createCartItemElement(item));
+  totalPrice();
+};
+
+function createProductItemElement({ sku, name, image }) { // Creating the list of available products
+  const section = document.createElement('section');
+  section.className = 'item';
+
+  section.appendChild(createCustomElement('span', 'item__sku', sku));
+  section.appendChild(createCustomElement('span', 'item__title', name));
+  section.appendChild(createProductImageElement(image));
+
+  const button = createCustomElement('button', 'item__add', 'Adicionar ao carrinho!'); // Adding event to product button to put this item in the cart
+  button.addEventListener('click', (e) => {
+    const clickedItemSku = e.target.parentNode.firstChild.innerText;
+
+    fetch(`https://api.mercadolibre.com/items/${clickedItemSku}`)
+    .then(response => response.json())
+    .then(json => newCartItem(json)) // Callback function to put clicked item in the cart
+    .then(() => toLocalStorage());
+  });
+
+  section.appendChild(button);
+  return section;
+}
+
+const productsList = (productArray) => {
+  setTimeout(() => {
+    document.querySelector('.loading').remove();
+  }, 3000);
+
+  const items = document.getElementsByClassName('items');
+  productArray.forEach((product) => {
+    const { id, title, thumbnail } = product;
```

Comment:
Ótima solução!! Aqui um outro jeito de fazer seria colocar o objeto diretamente dentro do parâmetro da função.
```suggestion
  productArray.forEach({ id, title, thumbnail } => {
```